src/main/resources/webapp/interface/Glue.es

```diff
@@ -91,11 +122,118 @@ function addUnionType({
 		typeResolver,
 		types
 	};
-	return this.unionTypes[name];
+	return this.unionTypes[name].type;
 }
 
 function getUnionType(name) {
-	return this.unionTypes[name];
+	if (!Object.hasOwn(this.unionTypes, name)) { // true also when property is set to undefined
+		if (this.uniqueNames[name]) {
+			throw new Error(`name:${name} is not an unionType! but ${this.uniqueNames[name]}`);
+		}
+		throw new Error(`name:${name} not found in unionTypes, perhaps you're trying to use it before it's defined?`);
+	}
+	return this.unionTypes[name].type;
+}
+
+/*function getUnionTypeResolver(name) {
+	return this.unionTypes[name].typeResolver;
+}*/
+
+
+function addInterfaceType({
+	fields,
+	name,
+	typeResolver
+}) {
+	if(this.uniqueNames[name]) {
+		throw new Error(`Name ${name} already used as ${this.uniqueNames[name]}!`);
+	}
+	this.uniqueNames[name] = 'interfaceType';
+	this.interfaceTypes[name] = {
+		fields,
+		type: this.schemaGenerator.createInterfaceType({
+			fields,
+			name,
+			typeResolver
+		})//,
+		//typeResolver // NOTE This should be fetched from the unionType instead...
+	};
+	return this.interfaceTypes[name].type;
+}
+
+function getInterfaceType(name) {
+	return this.interfaceTypes[name].type;
+}
+
+function getInterfaceTypeFields(name) {
+	return this.interfaceTypes[name].fields;
+}
+
+function getInterfaceTypeObject(name) {
+	return this.interfaceTypes[name];
+}
+
+
+function addQueryField({
+	args = {},
+	name,
+	resolve,// = () => {},
+	type
+}) {
+	if(this.queryFields[name]) {
+		throw new Error(`Name ${name} already added!`);
+	}
+	this.queryFields[name] = {
+		args,
+		resolve,
+		type
+	};
+	return this.queryFields[name];
+}
+
+
+function buildSchema() {
+	/*
+	 GIVEN that an objectType that implements an interface
+	 AND that objectType is NOT directly added/available in the schema
+	 WHEN a query result includes the interfaceType
+	 THEN GraphQL will throw an error that the objectType is not a valid type
+
+	 This happens because the interfaceType.typeResolver() returns an objectType
+	 that doesn't exist in the schema.
+	 Simply adding the objectType to the dictionary, solves the problem.
+	*/
+	const uniqObjectTypesWithInterfaces = [];
+	Object.keys(this.objectTypes).forEach((k) => {
+		if (
+			this.objectTypes[k].interfaces
+			&& Array.isArray(this.objectTypes[k].interfaces)
+			&& this.objectTypes[k].interfaces.length
+		) {
+			let alreadyAdded = false;
+			uniqObjectTypesWithInterfaces.forEach((obj) => {
+				if (obj === this.objectTypes[k].type) {
+					alreadyAdded = true; // TODO Break
+				}
+			});
+			if (!alreadyAdded) {
+				uniqObjectTypesWithInterfaces.push(this.objectTypes[k].type);
+			}
+		}
+	});
```

Comment:
How many items are we talking about here? You have `forEach` nested inside another `forEach` which might be crazy slow for big sets. 

I suggest making `uniqObjectTypesWithInterfaces` a hash map instead of an array. This way you wouldn't have to iterate over it to find if a type has already been added. 

This:
```
this.objectTypes[k].interfaces
			&& Array.isArray(this.objectTypes[k].interfaces)
			&& this.objectTypes[k].interfaces.length
```
should also be extracted to a helper method, it's not very pretty.

I also assume that if `this.objectTypes[k].type` is already in `uniqObjectTypesWithInterfaces` then there's no need for the check above? Not sure, it's too easy to get lost in the logic here. The example below assumes it's enough.

```
function hasInterfaces(objectType) {
  return objectType.interfaces
			&& Array.isArray(objectType.interfaces)
			&& objectType.interfaces.length;
}

const uniqObjectTypesWithInterfaces = {};
Object.keys(this.objectTypes).forEach((k) => {
  const objectType = this.objectTypes[k].type;
  if (uniqObjectTypesWithInterfaces[objectType]) {
    return;
  } 

  if (hasInterfaces(this.objectTypes[k]) {
    uniqObjectTypesWithInterfaces[objectType] = true;
  }
});
```

Reply:
Array vs Hash lookup speed differences aren't really much different before becomes in multiple thousands. But yeah, this code can certainly be improved a lot.